cpp/tests/orderby/is_sorted/is_sorted_tests.cu

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//  #include <tests/utilities/base_fixture.hpp>
+//  #include <gtest/gtest.h>
+//  #include <gmock/gmock.h>
+#include <cudf/cudf.h>
+#include <cudf/types.hpp>
+#include <tests/utilities/base_fixture.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/predicates.hpp>
+#include <cudf/sorting.hpp>
+
+//  #include <cudf/utilities/error.hpp>
+
+#include <cudf/column/column_factories.hpp>
+#include <tests/utilities/column_utilities.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <tests/utilities/type_lists.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+#include <tests/utilities/legacy/cudf_test_utils.cuh>
+#include <vector>
+
+
+template<typename T>
+using fixed_width_column_wrapper = cudf::test::fixed_width_column_wrapper<T>;
+
+template <typename T>
+struct IsSortedNumeric : public cudf::test::BaseFixture {};
+
+using test_types_numeric = ::testing::Types<int8_t, int16_t, int32_t, int64_t,
```

Comment:
This should use one of the type lists like `cudf::test::FixedWidthTypes`.

Reply:
@jrhemstad ... what if i wrote tests that work for both NumericTypes and std::string? would I have to duplicate those test definitions?

Reply:
`cudf::test::AllTypes`?

Reply:
`AllTypes` is tricky because you can't just uniformly use `fixed_width_column_wrapper<TypeParam>`, if `TypeParam == string_view` you need to use `string_column_wrapper`.

Reply:
I have tests which will work for both `fixed_width_column_wrapper` and `string_column_wrapper`, the only difference being:

```
TYPED_TEST_CASE(IsSortedNumeric, cudf::test::NumericTypes);
```
```
TYPED_TEST_CASE(IsSortedNumeric, ::testing::Types<std::string>);
```

I'm accomplishing this by injecting test data via explicit specialization. Syntactically, the tests for all data types are the same - the only difference being the data used in those tests.

Reply:
This approach also allows me to avoid special exceptions for specific types, like the one seen in the `sorted_order` tests when using `::experimental::bool8`.

https://github.com/rapidsai/cudf/blob/branch-0.11/cpp/tests/sort/sort_test.cpp#L52-L68

Reply:
I did a quick experiment to see what would happen when I removed fixed_width_column's static_assert for fixed-width types (which is what was preventing me from compiling). The code compiles, but some of my tests use empty vectors, and apparently our string column factory doesn't support zero length string columns... those tests failed, and I stopped digging at that point.

Reply:
I pushed the tests so you have can have a better idea what I'm talking about.